### PR TITLE
Pruner: Throwing errors when encountering NaN

### PR DESCRIPTION
**Pull request by lducas**
- throwing errors on NaN
- throwing error in infinite loop in descent_step
- additional printouts on flag PRUNER_VERBOSE

### CODE REVIEW

File: fplll/pruner.cpp [hunk 1]
```diff
@@ -289,6 +298,11 @@ inline FT Pruner<FT>::single_enum_cost(/*i*/ const evec &b, vector<double> *deta
     total += tmp;
     normalized_radius_pow *= normalized_radius;
   }
+  if (!total.is_finite())
+  {
+    throw std::runtime_error("NaN or inf in single_enum_cost");
```
Comment (malb) [inline fplll/pruner.cpp]: Some of these might be more appropriate http://en.cppreference.com/w/cpp/error/exception

Comment (lducas) [inline fplll/pruner.cpp]: hum. Would you consider this an overflow ?

Comment (malb) [inline fplll/pruner.cpp]: I was thinking that `is_finite` checks if the value is within representable range.

**Comment by lducas** [comment]
K

Le 10 août 2017 11:56, "Martin Albrecht" <notifications@github.com> a
écrit :

**Pull request merged**